src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -569,6 +571,53 @@ private bool DelegateNeedsReturn(MethodSymbol invokeMethod)
             return true;
         }
 
+        internal NamedTypeSymbol? InferDelegateType(ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
+        {
+            var compilation = Binder.Compilation;
+            var parameterRefKindsBuilder = ArrayBuilder<RefKind>.GetInstance(ParameterCount);
+            var parameterTypesBuilder = ArrayBuilder<TypeWithAnnotations>.GetInstance(ParameterCount);
+            for (int i = 0; i < ParameterCount; i++)
+            {
+                parameterRefKindsBuilder.Add(HasExplicitlyTypedParameterList ? RefKind(i) : default);
+                parameterTypesBuilder.Add(HasExplicitlyTypedParameterList ? ParameterTypeWithAnnotations(i) : TypeWithAnnotations.Create(compilation.GetSpecialType(SpecialType.System_Object)));
+            }
+            var parameterRefKinds = parameterRefKindsBuilder.ToImmutableAndFree();
```

Comment:
>parameterRefKindsBuilder [](start = 36, length = 24)

We generally use `default` if there are no RefKinds. Consider lazily allocating this builder instead.

Reply:
Unfortunately, `LambdaSymbol` expects a fully populated array.